src/components/utility/InvisibleDownload.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { forwardRef, useCallback, useImperativeHandle, useRef } from 'react'
+import { runDelayedSync } from '../../utils/delay/runDelayed'
+
+type InvisibleDownloadHandle = {
+  trigger: (options: { url: string }) => Promise<void>
+}
+
+export function useInvisibleDownload() {
+  const invisibleDownloadRef = useRef<InvisibleDownloadHandle>(null)
+
+  const triggerInvisibleDownload = useCallback((options: { url: string }) => {
+    invisibleDownloadRef.current?.trigger(options)
+  }, [])
+
+  return { invisibleDownloadRef, triggerInvisibleDownload }
+}
+
+const CLASS_NAME = 'Layer__InvisibleDownload'
+
+const InvisibleDownload = forwardRef<InvisibleDownloadHandle>((_props, ref) => {
```

Comment:
This is an intentional departure from `location.href =` 

I believe that this approach is (marginally) more defensible, since we're using the intended pathway for browser downloads. The former is a redirect that happens to produce a download.